### PR TITLE
fix: include the "types" entry in package.json files

### PR DESCRIPTION
**Pull request by Westbrook**
## Description 
Leverage the types entry in `package.json`.
Normalize the includes/excludes in `tsconfig.json`

## Motivation and Context
https://www.skypack.dev/view/@spectrum-web-components/bundle is now using the presence of types as a signifier of quality, and apparently even TS thinks this is the "right" way to do things: https://docs.skypack.dev/package-authors/package-checks#types

## How Has This Been Tested?
Hard to fully test, but seems like the right thing to do.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/accordion/package.json [hunk 1]
```diff
@@ -20,8 +20,9 @@
     ],
     "version": "0.1.4",
     "description": "",
-    "main": "src/index.js",
-    "module": "src/index.js",
+    "main": "./src/index.js",
+    "module": "./src/index.js",
+    "types": "./src/index.d.ts",
```
Comment (Westbrook) [inline packages/accordion/package.json:25]: Ensure types are referenced in `package.json`.

File: packages/icons-ui/tsconfig.json [hunk 2]
```diff
@@ -5,5 +5,7 @@
         "outDir": "./lib",
         "rootDir": "./src"
     },
-    "include": ["src/**/*.ts", "../../global.d.ts"]
+    "include": ["src/**/*.ts", "../../global.d.ts"],
+    "exclude": ["test/*.ts", "stories/*.ts"],
+    "references": [{ "path": "../base" }]
```
Comment (Westbrook) [inline packages/icons-ui/tsconfig.json:10]: Normalize include/exclude of files.

File: packages/base/src/Base.ts [hunk 3]
```diff
@@ -11,7 +11,10 @@ governing permissions and limitations under the License.
 */
 
 import { LitElement, property, UpdatingElement } from 'lit-element';
-import { Theme } from '@spectrum-web-components/theme';
+type ThemeRoot = HTMLElement & {
+    startManagingContentDirection: (el: HTMLElement) => void;
+    stopManagingContentDirection: (el: HTMLElement) => void;
+};
```
Comment (Westbrook) [inline packages/base/src/Base.ts:17]: Prevent cycle with local types.

File: test/tsconfig-test.json [hunk 4]
```diff
@@ -1,9 +1,9 @@
 {
     "extends": "../tsconfig.json",
     "compilerOptions": {
-        "composite": true,
         "outDir": "../",
-        "types": ["mocha", "chai", "sinon", "webpack-env", "node"]
+        "types": ["mocha", "chai", "sinon", "webpack-env", "node"],
+        "declaration": false
```
Comment (Westbrook) [inline test/tsconfig-test.json:6]: Not emitting declarations at testing times means that already build declarations from build time are not overwritten, which `tsc` throws on.

**Review by Westbrook — COMMENTED** [review]
A lot of repetitive changes, I've called out the major outliers.

**Comment by davidyxu** [comment]
changes look fine to me although the build is failing on `visual-light-large-ltr`, I assume this is an unrelated change though?

**Comment by Westbrook** [comment]
Yeah, a sometimes flaky test went mostly flaky there. I'm attempting to correct it right now in `westbrook/top-nav` and should have a separate PR soon.

**Comment by Westbrook** [comment]
@davidyxu upstream flakiness corrected!

**Review by davidyxu — APPROVED** [review]
lgtm! i noticed one little oddity in the `yarn.lock`, where the locked version of typescript reverted from `4.0.5` to `4.0.3`, probably because this pr has been floating around for awhile now. i doubt it'll cause any issues, but thought it was worth calling out.

**Pull request merged**